jlox

```diff
@@ -1,4 +1,14 @@
 #!/usr/bin/env bash
 
-script_dir=$(dirname "$0")
-java -cp ${script_dir}/build/java com.craftinginterpreters.lox.Lox $@
+if [[ "$OSTYPE" == "linux-gnu"* ]]
+then
+	DIR=$(dirname $(readlink -f "$0"))
+elif [[ "$OSTYPE" == "darwin"* ]]
+then
+	LINK=$(python -c "import os,sys; print(os.path.realpath(os.path.expanduser(sys.argv[1])))" "${0}")
```

Comment:
Disclaimer: This is pure banter, I have 0 experience building these types of cross-platform scripts.

It seems wild to invoke an entire python instance to expand a path :D Is this really standard practice?

Reply:
ah I would love to not use python, but macos doesn't come with a standard tool to fix this issue. I'd rather not force users to download gnu readlink or use a shell hack that might fail some times.

however, I could possibly add a check to ensure that the python instance is only called if the script is a symbolic link?

Reply:
By all means leave it as it is, I was just wondering (and enlightened) by the approach :)

Reply:
Isn't there anything better [here](https://stackoverflow.com/questions/1308823/os-x-bash-dirname)?